### DISCUSSION TITLE
A research for generating PR checklists in Pull Request Template

### DISCUSSION THREAD

**Discussion by chenkun1029** [react-hook-form/react-hook-form#11227] [Ideas]
Dear maintainers:
Our team is currently developing a tool for automatically generating PR checklists in Pull Request Template. The tool generates a set of PR rules based on the contribution guidelines in your project, which external contributors may be expected to adhere to. Below are the results of our generation:

**Test**: Test suite still passes.
**Documentation**: Ensure exports are documented
**Style**: Follow the coding style of the project
**Branch**: Create a new branch from master.

Also, we suggest you add some checklist notes about **Title**, **Description**, , **commit message**, **contributing**, and **cla**, which are also important for submitting PRs according to our analysis.

We hope you and your team can provide feedback on our generated output.
Best regards.

**Comment by foxsayno** [comment]
sounds interesting
